### PR TITLE
Fix the data generator element size for decimal types

### PR DESCRIPTION
**Pull request by vuule**
Data generator uses `sizeof` for all fixed width types and thus gets incorrect size for the decimal types. This leads to random tables that are actually half the requested size, causing incorrect benchmark bandwidth results.
This PR changes the `avg_element_size` to use `data_type::size_of`, which accounts for decimal types.

Note: expected to effectively half the BW of decimal benchmarks that specify the table size in bytes.

### CODE REVIEW

File: cpp/benchmarks/common/generate_input.cpp [hunk 1]
```diff
@@ -65,45 +65,46 @@ T get_distribution_mean(distribution_params<T> const& dist)
...
-    return avg_element_size<T>(profile);
+    return non_fixed_width_size<T>(profile);
   }
 };
 
-size_t avg_element_bytes(data_profile const& profile, cudf::type_id tid)
+size_t avg_element_size(data_profile const& profile, cudf::data_type dtype)
 {
-  return cudf::type_dispatcher(cudf::data_type(tid), avg_element_size_fn{}, profile);
+  if (is_fixed_width(dtype)) { return size_of(dtype); }
```
Comment (ttnghia) [inline cpp/benchmarks/common/generate_input.cpp:106]: Please use the prefix `cudf::` consistently: If already used it then always use it.

File: cpp/benchmarks/common/generate_input.cpp [hunk 2]
```diff
@@ -65,45 +65,46 @@ T get_distribution_mean(distribution_params<T> const& dist)
 }
 
 // Utilities to determine the mean size of an element, given the data profile
-template <typename T>
-std::enable_if_t<cudf::is_fixed_width<T>(), size_t> avg_element_size(data_profile const& profile)
+template <typename T, std::enable_if_t<cudf::is_fixed_width<T>()>* = nullptr>
```
Comment (mythrocks) [inline cpp/benchmarks/common/generate_input.cpp:68]: Nitpick: Is `CUDF_ENABLE_IF()` available at this point?
```suggestion
template <typename T, CUDF_ENABLE_IF(cudf::is_fixed_width<T>())>
```

**Review by mythrocks — APPROVED** [review]
Minuscule nitpick, but LGTM even otherwise.

**Comment by vuule** [comment]
@gpucibot merge

**Pull request merged**